core/src/test/java/org/elasticsearch/common/PriorityTests.java

```diff
@@ -82,11 +81,11 @@ public void testCompareTo() {
         for (Priority p : Priority.values()) {
             assertEquals(0, p.compareTo(p));
         }
-        List<Priority> shuffeledAndSorted = new ArrayList<>(Priority.values());
+        List<Priority> shuffeledAndSorted = Arrays.asList(Priority.values());
         Collections.shuffle(shuffeledAndSorted, random());
         Collections.sort(shuffeledAndSorted);
         for (List<Priority> priorities : Arrays.asList(shuffeledAndSorted,
-            Priority.values())) { // #values() guarantees order!
+            Arrays.asList(Priority.values()))) { // #values() guarantees order!
```

Comment:
Should we add a test to make sure the ordinal() is what we expect (i.e. IMMEDIATE is first, URGENT is second, etc) because now the declaration order of the enum values affects the comparison of two `Priority` values?